src/cockpit/channels/dbus.py

```diff
@@ -374,6 +383,19 @@ async def match_hit(message):
 
         self.add_async_signal_handler(match_hit, **add_match)
 
+    async def do_remove_match(self, message):
+        remove_match = message['remove-match']
+        logger.debug('remove match %s', remove_match)
+        r = self.get_r(**remove_match)
+        r_string = self.get_r_string(r)
+        for index, (r_key, slot) in enumerate(self.matches):
+            if r_key == r_string:
+                logger.debug('got match %s %d', self.matches, index)
+                slot.cancel()
```

Comment:
We need to do "use counting" for matches now. If "add-match" has been received 5 times, there need to be 5 matching "remove-match"es before the slot can be cancelled.  See protocol.md.